### DISCUSSION TITLE
Documentation on secretOrRandomPassword?

### DISCUSSION THREAD

**Discussion by pamelafox** [Azure/azure-dev#940] [Q&A] [answered]
This azd sample uses some sort of secretOrRandomPassword function in the parameters file:

https://github.com/Azure-Samples/todo-csharp-sql/blob/a10ba78f1322443f70a4ce36bcde93c47beaa3fe/infra/main.parameters.json

```
      "databasePassword": {
        "value": "$(secretOrRandomPassword ${AZURE_KEY_VAULT_NAME} databasePassword)"
      }
```

Is there any documentation on that? I haven't managed to find any online, but I think I also want to use that sort of functionality.

**Comment by pamelafox** [comment, marked as answer]
Okay I've figured out that it comes from here:
https://github.com/Azure/azure-dev/blob/c53a984deecf80d561cd0767a31e899a05e331db/cli/azd/pkg/cmdsubst/secretOrRandomPassword.go#L44

I think it would be helpful to document that. I'm happy to do the documentation, though I'm not sure where, since https://learn.microsoft.com/en-us/azure/developer/azure-developer-cli/reference is specifically about the CLI commands and this is about the parameters. It also seems that this special command is just for azd, so it doesnt belong in the generic bicep docs.

**Reply by jackreichelt** [reply]
Thank you, Pamela, this was very helpful! This still isn't documented anywhere!

**Reply by vhvb1989** [reply]
@weikanglim

**Reply by pamelafox** [reply]
There is documentation in the FAQ here now, from @alexwolfmsft :
https://learn.microsoft.com/en-us/azure/developer/azure-developer-cli/faq#what-is-the-behavior-of-the--secretorrandompassword--function

I'm guessing it's still hard to find? @jackreichelt Where were you looking for the docs? Maybe it belongs somewhere else.
